chapters/06-smart-contracts-in-ralph.md

```diff
@@ -1405,21 +1403,25 @@ import { OldCode, NewCode } from '../artifacts/ts'
 
 async function test() {
   const signer = await getSigner()
-  const { contractInstance: oldCode } = await OldCode.deploy(signer, {
-     initialFields: { owner: signer.address, n: 100n }
+  const { contractInstance: oldContract } = await OldCode.deploy(signer, {
+    initialFields: { owner: signer.address, n: 100n }
   })
 
-  await oldCode.transact.migrateWithFields({
+  await oldContract.transact.migrateWithFields({
     signer,
     args: { newCode: NewCode.contract.bytecode, newN: 200n }
   })
 
-  const newContract = await NewCode.at(oldCode.address)
-  const {returns: newN} = await newContract.view.get()
+  const newContract = NewCode.at(oldContract.address)
+  let newN = (await newContract.view.get()).returns
   console.assert(newN === 200n)
-  await newContract.transact.set({signer, args: {m: 300n}})
-  const {returns: newN2} = await newContract.view.get()
-  console.assert(newN2 === 300n)
+
+  await newContract.transact.set({ signer, args: { m: 300n } })
+  newN = (await newContract.view.get()).returns
+  console.assert(newN === 300n)
+
+  newN = (await oldContract.view.get()).returns
+  console.assert(newN === 300n)
```

Comment:
@h0ngcha0 do you agree with this addition? IMO, by comparing what we get when we call the `oldContract.view.get` function and confirming that it's what the `newContract.view.get` also returns, we crystalize that the migration has successfully took place.

Reply:
Yeah, that makes sense :+1: